### PR TITLE
Use custom database migrator service to ship Rails version-specific migrations

### LINKED ISSUE #2347: Failing migrations for mysql starting at migration CreateUploadedFiles
Branch:  master

### Descriptive summary

Migrations fail at CreateUploadedFiles when trying to create a new hyrax app using MySQL

### Rationale

Migrations should work for MySQL

### Expected behavior

All migrations should succeed.

### Actual behavior

CreateUploadedFiles migration fails for MySQL.  NOTE: It passes with engine_cart and sqlite.  It passes for postgres.

```
== 20171211132828 CreateUploadedFiles: migrating ==============================
-- adapter_name()
   -> 0.0000s
-- adapter_name()
   -> 0.0000s
-- create_table(:uploaded_files, {:id=>:integer})
rake aborted!
StandardError: An error has occurred, all later migrations canceled:

Mysql2::Error: Table 'hyrax_migrate_development.uploaded_files' doesn't exist: SHOW FULL FIELDS FROM `uploaded_files`
```

### Steps to reproduce the behavior

1. Remove `rake db:migrate` from hyrax generation template (https://gist.githubusercontent.com/elrayle/3250233bf21cec3f1086c44d56eefea2/raw/e185d17bd2f0936bae34ec0fd5c22653d57fc090/hyrax_new_template)
1. generate a new hyrax app with mysql as the db
```
rails new myapp -m https://gist.githubusercontent.com/elrayle/3250233bf21cec3f1086c44d56eefea2/raw/e185d17bd2f0936bae34ec0fd5c22653d57fc090/hyrax_new_template -d mysql
```
1. update config/database.yml to work with your mysql
1. rake db:drop     (if needed to clear out previous creation of the tables)
1. rake db:create db:migrate

Migrations should complete without errors.

### PR DESCRIPTION
**Pull request by mjgiarlo**
Fixes #2347

(Needs to be backported so a 2.0.1 release may be cut.)

@samvera/hyrax-code-reviewers

### CODE REVIEW

File: lib/tasks/install.rake [hunk 1]
```diff
@@ -0,0 +1,8 @@
+namespace :hyrax do
+  namespace :install do
+    desc 'Copy migrations from Hyrax to application'
+    task migrations: :environment do
```
Comment (mjgiarlo) [inline lib/tasks/install.rake]: This task is added to replace the built-in task of the same name that was automatically added to Hyrax until the changes made in this PR. It should work in the same way. (The other task, provided by Railties, does not handle ERB evaluation.)

**Comment by mjgiarlo** [comment]
Heads-up, @elrayle

**Comment by mjgiarlo** [comment]
@elrayle can you run through your test steps again (pointing the Gemfile at `gem 'hyrax', github: 'samvera/hyrax', branch: 'use_current_rails_version_for_migrations_2347'`) to make sure this fixes things you were seeing?

**Comment by mjgiarlo** [comment]
@jcoyne @aaron-collier This is ready for final review.

**Comment by elrayle** [comment]
@mjgiarlo The manual steps (not using the template) worked smoothly.  The migration completes without errors.  :1st_place_medal: 

The only error I see occurs while running `rails g hyrax:install -f`

```
error  Use the mysql table based generator if you are using mysql 'rails generate qa:local:tables:mysql'
```

But this is not in the migration process itself.  So it's really a different, but related issue.

**Pull request merged**